tests/periph_i2c/main.c

```diff
@@ -25,265 +25,525 @@
 #include "periph/i2c.h"
 #include "shell.h"
 
+#ifndef GET_I2C_CLK_SPD
+#define GET_I2C_CLK_SPD(x)  (-1)
+#endif
+
+
+#define INVALID_ARGS    puts("Error: Invalid number of arguments");
+
 #define BUFSIZE        (128U)
 
 static int i2c_dev = -1;
 
-static i2c_speed_t map_speed[] = {
-    I2C_SPEED_LOW,
-    I2C_SPEED_NORMAL,
-    I2C_SPEED_FAST,
-    I2C_SPEED_FAST_PLUS,
-    I2C_SPEED_HIGH
-};
 
-int cmd_init_master(int argc, char **argv)
-{
-    int dev, speed, res;
+int check_dev(void) {
+    if (i2c_dev < 0) {
+        puts("Error: no I2C device was initialized");
+        return 1;
+    }
+    return 0;
+}
+
+int print_i2c_error(int res) {
+    if (res == I2C_ADDR_NACK) {
+        puts("Error: addr NACK");
+        return 1;
+    }
+    else if (res == I2C_DATA_NACK) {
+        puts("Error: data NACK");
+        return 1;
+    }
+    else if (res == I2C_ERR) {
+        puts("Error: i2c return error");
+        return 1;
+    }
+    else if (res == I2C_ACK) {
+        printf("Success: Command successful\n");
+        return 0;
+    }
+    puts("Error: Unknown Error");
+    return 1;
+}
+
+int cmd_i2c_assign_dev(int argc, char **argv) {
+    int dev;
 
-    if (argc != 3) {
-        puts("Error: Init: Invalid number of arguments!");
-        printf("Usage:\n%s: [DEVICE] [SPEED]\n", argv[0]);
+    if (argc != 2) {
+        INVALID_ARGS;
+        printf("Usage:\n%s: [DEVICE]\n", argv[0]);
         puts("    with DEVICE:");
         for (unsigned i = 0; i < I2C_NUMOF; i++) {
-            printf("          %u -> I2C_%u\n", i, i);
+            printf("          %u -> i2c_%u\n", i, i);
         }
-        puts("         SPEED:");
-        puts("          0 -> SPEED_LOW (10kbit/s)");
-        puts("          1 -> SPEED_NORMAL (100kbit/s)");
-        puts("          2 -> SPEED_FAST (400kbit/s)");
-        puts("          3 -> SPEED_FAST_PLUS (1Mbit/s)");
-        puts("          4 -> SPEED_HIGH (3.4Mbit/s)\n");
         return 1;
     }
-
     dev = atoi(argv[1]);
-    speed = atoi(argv[2]);
-
-    if (speed < 0 || speed > 4) {
-        puts("Error: Speed value not in range (0 - 4)");
+    if (dev < 0 || dev >= (int)I2C_NUMOF) {
+        puts("Error: Given device not available");
+        i2c_dev = -1;
         return 1;
     }
+    else {
+        printf("Success: i2c_%i successfully assigned as master\n", dev);
+        i2c_dev = dev;
+        return 0;
+    }
+    return 1;
+}
+
+int cmd_i2c_acquire(int argc, char **argv) {
```

Comment:
Yup I know, I think if it is in the API it should be tested (the fact that it blocks shouldn't be a problem for the test since I should be able to expect timeout and provide resets).  If you don't wish to have this tested in the automated test we can remove it but I think since it is exposed to the user it should be tests.  Also some clarification on it's expected behavior would be nice (ie is it required to acquire before use? Can we release twice without getting an error?)

Reply:
You're right I mess up things sorry. acquire doesn't belong to CPU driver but device driver so we must be able to test the acquire/release function. 
You can leave it as it.